File: Java/src/Main/Bread.java
package Main;

public class Bread extends BakedGood {
    private final String variety;

    public Bread(String variety) {
        super(4.50);
        this.variety = variety;
    }

    @Override
    public String ToString() {
        return variety + " bread " + super.ToString();
    }

    @Override
    public double DiscountedPrice(int quantity) {
        return (quantity - quantity/3) * super.getBasePrice();
    }
}

File: Python3/Bread.py
from BakedGood import BakedGood

class Bread(BakedGood):
    def __init__(self, variety):
        super().__init__(4.50)
        self._variety = variety

    def __str__(self):
        return self._variety + " bread " + super().__str__()

    #Overriding abstract DiscountedPrice(quantity) from BakedGood
    def DiscountedPrice(self, quantity):
        return (quantity - quantity//3) * super().getBasePrice()
